### PR TITLE
Dockerize application

### PR DESCRIPTION
**Pull request by SlavikN14**
@ajax-shulepa-a 
@ajax-gnatiuk-s 
@VolodymyrM27

### CODE REVIEW

File: src/main/resources/application-docker.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+spring:
+  autoconfigure:
+    exclude: org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration
+  data:
+    mongodb:
+      uri: mongodb://mongo:27017/telegram-bot-db
+bot:
+  token: 6565348709:AAES0cSZrxrspwIDy6zzVhnCtRwMWveoYYc
```
Comment (VolodymyrM27) [inline src/main/resources/application-docker.yaml]: Never push tokens from third-party resources to the github, as there are bots that scan all open repositories and steal them

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+services:
+  telegram-bot:
+    build:
+      dockerfile: Dockerfile
+    container_name: "app"
+    ports:
+      - "8080:8080"
+    depends_on:
+      - mongo
```
Comment (VolodymyrM27) [inline docker-compose.yaml]: Tip: you can add a condition to check if the database container is okay - read more about healthcheck

**Review by VolodymyrM27 — APPROVED** [review]

File: src/main/resources/application-docker.yaml (hunk 1 above)
Comment (SlavikN14) [reply src/main/resources/application-docker.yaml]: My bad, i forgot to hide it in the new profile

File: src/main/resources/application-docker.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+spring:
+  autoconfigure:
+    exclude: org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration
+  data:
+    mongodb:
+      uri: mongodb://mongo:27017/telegram-bot-db
```
Comment (ajax-shulepa-a) [inline src/main/resources/application-docker.yaml]: will it work with your docker compose?

Comment (SlavikN14) [reply src/main/resources/application-docker.yaml]: Yes, it's working

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+services:
+
+  telegram-bot:
+    build:
+      dockerfile: Dockerfile
+    container_name: "app"
+    ports:
+      - "8080:8080"
+    depends_on:
```
Comment (VolodymyrM27) [inline docker-compose.yaml:11]: depends_on:
      mongodb:
        condition: service_healthy
        
A little advice on how to improve it

**Review by VolodymyrM27 — APPROVED** [review]

**Review by VolodymyrM27 — APPROVED** [review]

File: src/main/resources/application-docker.yaml (hunk 3 above)
Comment (ajax-gnatiuk-s) [reply src/main/resources/application-docker.yaml]: @ajax-shulepa-a checked yesterday on 1-2-1. Everything is ok here.

Why you doubt

File: src/main/resources/application-docker.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  data:
+    mongodb:
+      uri: mongodb://mongo:27017/telegram_bot_db
+
+bot:
+  token: #secret
+  username: BankInfoBot
+
+text:
+  path: /texts/texts.properties
```
Comment (ajax-gnatiuk-s) [inline src/main/resources/application-docker.yaml:13]: already defined in application.yaml
```suggestion
```

File: src/main/resources/application-docker.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+mono:
+  url: https://api.monobank.ua/bank/currency
```
Comment (ajax-gnatiuk-s) [inline src/main/resources/application-docker.yaml:16]: already defined in application.yaml

File: src/main/resources/application-docker.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+spring:
+  autoconfigure:
+    exclude: org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration
```
Comment (ajax-gnatiuk-s) [inline src/main/resources/application-docker.yaml:3]: already defined in application.yaml

File: src/main/resources/application-docker.yaml [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  data:
+    mongodb:
+      uri: mongodb://mongo:27017/telegram_bot_db
+
+bot:
+  token: #secret
+  username: BankInfoBot
```
Comment (ajax-gnatiuk-s) [inline src/main/resources/application-docker.yaml:10]: already defined in application.yaml

File: src/main/kotlin/com/ajaxproject/telegrambot/bot/utils/TextService.kt [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        val properties = Properties()
+        val resource = ClassPathResource(textProperties.path)
+        val inputStream = resource.inputStream
+
+        inputStream.use {
+            properties.load(inputStream)
+        }
+
+        return properties.getProperty(property)
+            ?: throw FileNameNotFoundException("File ${textProperties.path} not found")
```
Comment (ajax-gnatiuk-s) [inline src/main/kotlin/com/ajaxproject/telegrambot/bot/utils/TextService.kt:24]:
```suggestion
        return ClassPathResource(textProperties.path).inputStream
            .use { inputStream -> 
                Properties().apply { load(inputStream) }
            }
            .getProperty(property)
            ?: throw FileNameNotFoundException("File ${textProperties.path} not found")
```

**Review by ajax-gnatiuk-s — APPROVED** [review]

**Pull request merged**